pki/deploy.go

```diff
@@ -16,38 +16,38 @@ import (
 	"k8s.io/client-go/util/cert"
 )
 
-func convertCrtToENV(name string, certificate *x509.Certificate) string {
+func ConvertCrtToENV(name string, certificate *x509.Certificate) string {
 	encodedCrt := cert.EncodeCertPEM(certificate)
 	return fmt.Sprintf("%s=%s", name, string(encodedCrt))
 }
 
-func convertKeyToENV(name string, key *rsa.PrivateKey) string {
+func ConvertKeyToENV(name string, key *rsa.PrivateKey) string {
 	encodedKey := cert.EncodePrivateKeyPEM(key)
 	return fmt.Sprintf("%s=%s", name, string(encodedKey))
 }
 
-func convertConfigToENV(name string, config string) string {
+func ConvertConfigToENV(name string, config string) string {
 	return fmt.Sprintf("%s=%s", name, config)
 }
 
 func DeployCertificatesOnMasters(cpHosts []hosts.Host, crtMap map[string]CertificatePKI) error {
 	env := []string{
-		convertCrtToENV(CACertENVName, crtMap[CACertName].Certificate),
-		convertKeyToENV(CAKeyENVName, crtMap[CACertName].Key),
-		convertCrtToENV(KubeAPICertENVName, crtMap[KubeAPICertName].Certificate),
-		convertKeyToENV(KubeAPIKeyENVName, crtMap[KubeAPICertName].Key),
-		convertCrtToENV(KubeControllerCertENVName, crtMap[KubeControllerName].Certificate),
-		convertKeyToENV(KubeControllerKeyENVName, crtMap[KubeControllerName].Key),
-		convertConfigToENV(KubeControllerConfigENVName, crtMap[KubeControllerName].Config),
-		convertCrtToENV(KubeSchedulerCertENVName, crtMap[KubeSchedulerName].Certificate),
-		convertKeyToENV(KubeSchedulerKeyENVName, crtMap[KubeSchedulerName].Key),
-		convertConfigToENV(KubeSchedulerConfigENVName, crtMap[KubeSchedulerName].Config),
-		convertCrtToENV(KubeProxyCertENVName, crtMap[KubeProxyName].Certificate),
-		convertKeyToENV(KubeProxyKeyENVName, crtMap[KubeProxyName].Key),
-		convertConfigToENV(KubeProxyConfigENVName, crtMap[KubeProxyName].Config),
-		convertCrtToENV(KubeNodeCertENVName, crtMap[KubeNodeName].Certificate),
-		convertKeyToENV(KubeNodeKeyENVName, crtMap[KubeNodeName].Key),
-		convertConfigToENV(KubeNodeConfigENVName, crtMap[KubeNodeName].Config),
+		ConvertCrtToENV(CACertENVName, crtMap[CACertName].Certificate),
```

Comment:
We should think about creating cert/keys envvars in some more generic manner. Like defining 2 slices, one with the certNames, another with the keyNames, and parsing them accordingly.